Removing more mentions of .packages (#3966)
dart-lang/site-www: src/tools/pub/package-layout.md

{% prettify none tag=pre+code %}
enchilada/
  .dart_tool/ *

>   package_config.json *

Reviewer: The `package_config.json` file is part of the  `.dart_tool/` directory, so I don't think this is needed:

```suggestion
```

    style.css
{% endprettify %}

> {{site.alert.secondary}}

Reviewer: ```suggestion
{{site.alert.version-note}}
```

  **Version note:** Prior to Dart 2.17, `pub get` also
  generated a `.packages` file. That generated file has been
  replaced by `package_config.json`.
{{site.alert.end}}

> \* The `.dart_tool/` directory and `package_config.json`
>    files exist after you've run `pub get`.

Reviewer: This one we could mention `package_config.json` but I'm not sure it is necessary since it is part of the `.dart_tool/` directory.

```suggestion
\* The `.dart_tool/` directory exists after you've run `pub get`. 
   Don't check it into source control.
```

> \* The `.dart_tool/` directory and `package_config.json`
>    files exist after you've run `pub get`.
>    Don't check them into source control.

Reviewer: This one we could mention `package_config.json` but I'm not sure it is necessary since it is part of the `.dart_tool/` directory.

```suggestion
\* The `.dart_tool/` directory exists after you've run `pub get`.
   Don't check it into source control.
```
